subiquity/server/controllers/mirror.py

```diff
@@ -156,9 +199,51 @@ async def POST(self, data: str):
         self.model.set_mirror(data)
         await self.configured()
 
+    async def candidate_POST(self, url: str) -> None:
+        log.debug(url)
+        self.model.set_mirror(url)
+
     async def disable_components_GET(self) -> List[str]:
-        return self.model.config.get('disable_components', [])
+        return sorted(self.model.disabled_components)
 
     async def disable_components_POST(self, data: List[str]):
         log.debug(data)
-        self.model.config['disable_components'] = data
+        self.model.disabled_components = set(data)
+
+    async def check_mirror_start_POST(
+            self, cancel_ongoing: bool = False) -> None:
+        if self.mirror_check is not None and not self.mirror_check.task.done():
+            if cancel_ongoing:
+                await self.check_mirror_abort_POST()
+            else:
+                assert False
```

Comment:
er. This probably shouldn't still be here?

Reply:
Right, fixed!